### PR TITLE
[ADD] template: PhantomJS Test Sample

### PR DESCRIPTION
**Pull request by lasley**
### Milestone (Odoo version)
- 9.0
### Module(s)
- Template
### Fixes / new features
- Adds an example for QUnit test suite injection via QWeb
- Adds PhantomJS test example

### CODE REVIEW

**Comment by pedrobaeza** [comment]
I think we should have branches 8.0, 9.0 of MT to accommodate all of these changes to all versions. Can you please bring the topic to mailing list?

About this, isn't needed a Python file to launch the test?

**Comment by lasley** [comment]
@pedrobaeza you are correct that a Python file is also required - we already had that part in our template for some reason. 

Writing the email now

File: template/module/static/tests/js/module_name.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/* Copyright <YEAR(S)> <AUTHOR(S)>
+ * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). */
+
+odoo.define_section('module_name', ['module_name.ExportedObject'], function(test) {
+    "use strict";
+
+    test('It should demonstrate a PhantomJS test',
```
Comment (Yajo) [inline template/module/static/tests/js/module_name.js]: Please specify that this is for backend tests.

File: template/module/static/tests/js/module_name.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        function(assert, ExportedObject) {
+            var expect = 'Expected Return',
+                result = new ExportedObject();
+            assertStrictEqual(
```
Comment (Yajo) [inline template/module/static/tests/js/module_name.js]: this must be `assert.assertStrictEqual`, right?

**Comment by lasley** [comment]
Thanks @Yajo - updated

**Comment by oscarolar** [comment]
👍

**Pull request merged**